### PR TITLE
Write opt

### PR DESCRIPTION
**Pull request by bartliu827**
# Which issue does this PR close?
#591 
implement read iterator for query

Closes #.

# Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Review by roseboy-liu — APPROVED** [review]
LGTM

File: query_server/query/src/iterator.rs [hunk 1]
```diff
@@ -0,0 +1,621 @@
...
+    fn record_builder(&self) -> Vec<ArrayBuilderPtr> {
+        let mut builders: Vec<ArrayBuilderPtr> =
+            Vec::with_capacity(self.option.table_schema.fields.len());
+        for (_, item) in self.option.table_schema.fields.iter() {
+            debug!("schema info {:02X} {}", item.id, item.name);
+
+            match item.column_type {
+                ColumnType::Tag => builders.push(Box::new(StringBuilder::new(self.batch_size))),
+                ColumnType::Time => {
+                    builders.push(Box::new(TimestampNanosecondBuilder::new(self.batch_size)))
```
Comment (roseboy-liu) [inline query_server/query/src/iterator.rs:551]: Data Type

Comment (roseboy-liu) [reply query_server/query/src/iterator.rs:551]: TimestampNanosecondBuilder

File: query_server/query/src/stream.rs [hunk 2]
```diff
@@ -36,102 +31,92 @@ enum ArrayType {
...
+            table_schema.db.clone(),
+            table_schema.name.clone(),
+            proj_fileds,
+        );
+
+        let option = QueryOption {
+            time_range: TimeRange {
+                min_ts: i64::MIN,
+                max_ts: i64::MAX,
+            },
```
[the remark this answers is not in the document]
Comment (bartliu827) [reply query_server/query/src/stream.rs]: done

**Pull request merged**